File: frontend/campus-erp-ui/src/app/core/models/create-student-request.ts
export interface CreateStudentRequest {
  institutionId: string;
  campusId: string;
  departmentId: string;
  courseId: string;

  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;

  password: string;

  rollNumber: string;
  batch: string;
  admissionDate: string;
}

File: backend/CampusERP.Contracts/Requests/CreateStudentRequest.cs
using CampusERP.Contracts.Enums;

namespace CampusERP.Contracts.Requests;

public class CreateStudentRequest
{
    public Guid InstitutionId { get; set; }

    public Guid CampusId { get; set; }

    public Guid DepartmentId { get; set; }

    public Guid CourseId { get; set; }

    public Guid SemesterId { get; set; }

    public Guid SectionId { get; set; }

    public string AdmissionNumber { get; set; } = string.Empty;

    public string FirstName { get; set; } = string.Empty;

    public string LastName { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string? PhoneNumber { get; set; }

    public string Password { get; set; } = string.Empty;

    public string RollNumber { get; set; } = string.Empty;

    public string Batch { get; set; } = string.Empty;

    public DateTime AdmissionDate { get; set; }
}